service/header/core_listener_test.go

```diff
@@ -50,16 +54,40 @@ func TestCoreListener(t *testing.T) {
 }
 
 func createMocknetWithTwoPubsubEndpoints(t *testing.T) (*pubsub.PubSub, *pubsub.PubSub) {
-	host1, host2 := createMocknet(context.Background(), t)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	t.Cleanup(cancel)
```

Comment:
Can you pls make this context being global and defined in the test? So all the places where context is accepted take the same context and backgrounds.

Reply:
Yes